src/manual.ts

```diff
@@ -48,6 +50,19 @@ init();
 
 // Converts manual into the opposite format
 function renderManual(step?: string | (() => void)) {
+
+  /**
+   * Generate Table of Contents here, must not be rebasing.
+   */
+  if (!Git.rebasing()) {
+    const log = [ Git(['--no-pager', 'log', '--format=%s'], { cwd: Git.getCWD() }) ]
+      .map(str => str.split('\n'))
+      .map(arr => JSON.stringify(arr, null, 4))
+      .pop();
+
+    LocalStorage.setItem('TABLE_OF_CONTENTS', log);
```

Comment:
Why do you use the local storage? So you won't have to run it every time? I think that storing it in-memory somehow can be a more appropriate solution, less heavier. If not, can you add a comment in-code?

Reply:
@DAB0mB I use LocalStorage because tortilla runs several processes recursively to render all steps. I tried the way you said before but only worked when rendering the root template, and cause errors when rendering all of them. I also could not find a way to store in memory so it would run only for the root template and ignore other templates... it seems when we know we should get the log, the repository is well within rebasing it... and I guess we should allow users to place the Table of Contents elsewhere too. 

That is a valid concern, glad you asked, if there is a better solution please point it out to me.